### PR TITLE
cartridge improvements

### PR DESCRIPTION
**Pull request by asad-rafter**

### CODE REVIEW

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml [hunk 1]
```diff
@@ -6,7 +6,6 @@
 <isset name="sitePref" value="${dw.system.Site.current.preferences.custom}" scope="page" />
 <isscript>
     // SFRA has a logic bug where full urls are added twice, so we need to check for duplicates before adding.
-    var assets = require('*/cartridge/scripts/assets.js');
     pdict.viewmodel.CLDPDVideoPlayerJSURL = sitePref.CLDPDVideoPlayerJSURL;
     pdict.viewmodel.CLDPDShrinkwrapJSURL = sitePref.CLDPDShrinkwrapJSURL;
```
Comment (yuval-cloudinary) [inline cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: please remove all references to the shrinkwrap lib, as it is not used anywhere

**Review by yuval-cloudinary — CHANGES_REQUESTED** [review]

File: cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibraryVideo.js [hunk 2]
```diff
@@ -1,9 +1,6 @@
 'use strict';
 
-var Template = require('dw/util/Template');
-var HashMap = require('dw/util/HashMap');
 var currentSite = require('dw/system/Site').getCurrent();
```
[the remark this answers is not in the document]
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibraryVideo.js:3]: It is used in multiple functions, that's why  we make it global.

File: cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibrary.js [hunk 3]
```diff
@@ -1,13 +1,6 @@
 'use strict';
 
-var Template = require('dw/util/Template');
-var HashMap = require('dw/util/HashMap');
 var currentSite = require('dw/system/Site').getCurrent();
```
[the remark this answers is not in the document]
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibrary.js:3]: It is used in multiple functions, that's why we make it global.

File: cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibraryVideo.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibraryVideo.js:3]: It is still a performance penalty for functions that don't use it. Please have it only where it is needed, in the smallest scope that makes sense

File: cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibraryVideo.js [hunk 4]
```diff
@@ -1,9 +1,6 @@
...
-var log = require('dw/system').Logger.getLogger('Cloudinary', '');
 var constants = require('~/cartridge/experience/utils/cloudinaryPDConstants').cloudinaryPDConstants;
```
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibraryVideo.js:4]: It is still a performance penalty for functions that don't use it. Please have it only where it is needed, in the smallest scope that makes sense

File: cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibrary.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibrary.js:3]: It is still a performance penalty for functions that don't use it. Please have it only where it is needed, in the smallest scope that makes sense

File: cartridges/int_cloudinary_pd/cartridge/static/default/js/cloudinaryVideos.js [hunk 5]
```diff
@@ -20,24 +20,22 @@ function initializeCloudinaryPlayers() {
...
-            initializeCloudinaryPlayers();
-        }, 100);
+        let conf = {
+            cloud_name: window.cloudName
+        }
+        if (window.cname) {
+            conf.secure_distribution = window.cname;
+            conf.private_cdn = true;
+        }
+        cld = cloudinary.Cloudinary.new(conf);
```
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/static/default/js/cloudinaryVideos.js:38]: window.cld of course.
also check for other locations where 'cld' is used without explicit binding

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 1 above)
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: we have added shrink wrap recently as part of this 
https://github.com/cloudinary/cloudinary_sfcc_pagedesigner/pull/80/commits/b12f2f08c4316550f8255e5515909163c361a943

Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: which JIRA issue was adding the shrinkwrap?
PageDesigner itself doesn't use it for sure, maybe it was needed for the old video player?

File: cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibrary.js [hunk 6]
```diff
@@ -80,6 +73,8 @@ function generateBreakPoints(viewmodel) {
 }
```
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/experience/components/assets/mediaLibrary.js:75]: still open

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 1 above)
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: still open

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml [hunk 7]
```diff
@@ -6,7 +6,6 @@
 <isset name="sitePref" value="${dw.system.Site.current.preferences.custom}" scope="page" />
 <isscript>
     // SFRA has a logic bug where full urls are added twice, so we need to check for duplicates before adding.
-    var assets = require('*/cartridge/scripts/assets.js');
     pdict.viewmodel.CLDPDVideoPlayerJSURL = sitePref.CLDPDVideoPlayerJSURL;
```
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:9]: still open

[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:9]: @asad-rafter It seems you answered another question. Here, I asked "why is the videoplayer defined in the cloudinaryImage.ISML?"

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 1 above)
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: cld-304 has PR#81, not PR#80
JIRA has links to comments, can you please share the link to the comment on JIRA?

Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: In addition, the commit you shared leads me to PR#78

Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: image

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 7 above)
[the remark this answers is not in the document]
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:9]: We are getting the video player URL to compare with shrink wrap. If the video player URL doesn't exist only in that case we are including shrink wrap URL in DOM, at line 23

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 1 above)
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: image

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 7 above)
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:9]: Thanks, so please remove the shrinkwrap and the check for the Video Player, and deploy

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 1 above)
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: Thanks, so please remove the shrinkwrap and we will test it again

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 7 above)
[the remark this answers is not in the document]
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:9]: sure

File: cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml (hunk 1 above)
Comment (asad-rafter) [reply cartridges/int_cloudinary_pd/cartridge/templates/default/experience/components/assets/cloudinaryImage.isml:10]: sure

**Review by yuval-cloudinary — APPROVED** [review]

**Pull request merged**